pkg/controller/csi_handler.go

```diff
@@ -354,6 +357,22 @@ func (h *csiHandler) hasVAFinalizer(va *storage.VolumeAttachment) bool {
 	return false
 }
 
+// Checks if the PV (or) the inline-volume corresponding to the VA could have migrated from
+// in-tree to CSI.
+func (h *csiHandler) isMigratable(va *storage.VolumeAttachment) bool {
+	if va.Spec.Source.PersistentVolumeName != nil {
+		pv, err := h.pvLister.Get(*va.Spec.Source.PersistentVolumeName)
+		if err != nil {
+			return false
+		}
+		return h.translator.IsPVMigratable(pv)
+	} else if va.Spec.Source.InlineVolumeSpec != nil {
+		return va.Spec.Source.InlineVolumeSpec.CSI != nil
```

Comment:
@msau42: I am not sure if this is the right check for inline vols. Please suggest.

Reply:
It looks like the right check.